gradle-jdks/src/main/java/com/palantir/gradle/jdks/ToolchainsPlugin.java

```diff
@@ -58,12 +67,32 @@ public void apply(Project rootProject) {
         JdksExtension jdksExtension = JdksPlugin.extension(rootProject, jdkDistributions);
 
         rootProject.getPluginManager().withPlugin("com.palantir.baseline-java-versions", unused -> {
-            rootProject
-                    .getExtensions()
-                    .getByType(BaselineJavaVersionsExtension.class)
-                    .getSetupJdkToolchains()
-                    .set(false);
+            BaselineJavaVersionsExtension baselineJavaVersionsExtension =
+                    rootProject.getExtensions().getByType(BaselineJavaVersionsExtension.class);
+            baselineJavaVersionsExtension.getSetupJdkToolchains().set(false);
+
+            jdksExtension.jdkMajorVersionsToUse().set(rootProject.provider(() -> Stream.of(
```

Comment:
Feels strange to `.set` here and `.add` below in the subprojects. I would probably just `.add` here too.